Mathlib/Data/Matroid/Closure.lean

```diff
@@ -629,4 +631,115 @@ lemma ext_closure {M₁ M₂ : Matroid α} (h : ∀ X, M₁.closure X = M₂.clo
   eq_of_indep_iff_indep_forall (by simpa using h univ)
     (fun _ _ ↦ by simp_rw [indep_iff_forall_closure_diff_ne, h])
 
+
+section Spanning
+
+variable {S T I B : Set α}
+
+/-- A set is `spanning` in `M` if its closure is equal to `M.E`, or equivalently if it contains
+  a base of `M`. -/
+@[mk_iff]
+structure Spanning (M : Matroid α) (S : Set α) : Prop where
+  closure_eq : M.closure S = M.E
+  subset_ground : S ⊆ M.E
+
+attribute [aesop unsafe 10% (rule_sets := [Matroid])] Spanning.subset_ground
+
+lemma spanning_iff_closure (hS : S ⊆ M.E := by aesop_mat) : M.Spanning S ↔ M.closure S = M.E := by
```

Comment:
Weak suggestion, since I could see someone searching for `closure_eq`
```suggestion
lemma spanning_iff_closure_eq (hS : S ⊆ M.E := by aesop_mat) : M.Spanning S ↔ M.closure S = M.E := by
```